src/envoy/auth/jwt.cc

```diff
@@ -90,17 +102,48 @@ const uint8_t *CastToUChar(const std::string &str) {
   return reinterpret_cast<const uint8_t *>(str.c_str());
 }
 
+bssl::UniquePtr<EVP_PKEY> EvpPkeyFromRsa(RSA *rsa) {
+  if (!rsa) {
+    return nullptr;
+  }
+  bssl::UniquePtr<EVP_PKEY> key(EVP_PKEY_new());
+  EVP_PKEY_set1_RSA(key.get(), rsa);
+  return key;
+}
+
 bssl::UniquePtr<EVP_PKEY> EvpPkeyFromStr(const std::string &pkey_pem) {
   std::string pkey_der = Base64::decode(pkey_pem);
-  bssl::UniquePtr<RSA> rsa(
-      RSA_public_key_from_bytes(CastToUChar(pkey_der), pkey_der.length()));
+  return EvpPkeyFromRsa(
+      bssl::UniquePtr<RSA>(
+          RSA_public_key_from_bytes(CastToUChar(pkey_der), pkey_der.length()))
+          .get());
+}
+
+bssl::UniquePtr<RSA> RsaFromJwk(const std::string &n, const std::string &e) {
+  bssl::UniquePtr<RSA> rsa(RSA_new());
   if (!rsa) {
+    // Couldn't create RSA key.
     return nullptr;
   }
-  bssl::UniquePtr<EVP_PKEY> key(EVP_PKEY_new());
-  EVP_PKEY_set1_RSA(key.get(), rsa.get());
+  auto BigNumFromBase64UrlString = [](const std::string &s) -> BIGNUM * {
```

Comment:
Ugh, sorry I meant lambda doesn't make it clearer, and this is inefficient since you're making an function object every time.